tests/integration_test.rs

```diff
@@ -1,43 +1,172 @@
 use hyper::header::HOST;
+use lazy_static::lazy_static;
+use std::net::TcpListener;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Mutex;
 use std::{thread, time};
 use wiremock::matchers::{method, path};
 use wiremock::{Mock, MockServer, ResponseTemplate};
 
 use brachyura::run_server;
+pub struct MockBackend {
+    pub mock_server: Option<MockServer>,
+}
 
-#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
-async fn end_to_end_test() {
-    // Setup the mock server
-    let listener = std::net::TcpListener::bind("127.0.0.1:8000").unwrap();
-    let mock_server = MockServer::builder().listener(listener).start().await;
-    let body = "This is the mock server!";
-    let template = ResponseTemplate::new(200).set_body_raw(body, "text/plain");
-
-    Mock::given(method("GET"))
-        .and(path("/test"))
-        .respond_with(template)
-        .mount(&mock_server)
-        .await;
+impl MockBackend {
+    pub fn new() -> Self {
+        Self { mock_server: None }
+    }
+
+    pub async fn init(&mut self) {
+        let listener = TcpListener::bind("127.0.0.1:8000").unwrap();
+        let mock_server = MockServer::builder().listener(listener).start().await;
+        let template =
+            ResponseTemplate::new(200).set_body_raw("This is the mock backend!", "text/plain");
+
+        Mock::given(method("GET"))
+            .and(path("/test"))
+            .respond_with(template)
+            .mount(&mock_server)
+            .await;
+        self.mock_server = Some(mock_server);
+    }
+}
+
+lazy_static! {
+    pub static ref MOCK_BACKEND: Mutex<MockBackend> = Mutex::new(MockBackend::new());
+    pub static ref PROXY_STARTED: AtomicBool = AtomicBool::new(false);
+}
```

Comment:
Would recommend to try [once_cell](https://crates.io/crates/once-cell) instead of lazy_static. lazy_static is unformally deprecated.
Also `once_cell` gonna be stabilized in [stdlib](https://doc.rust-lang.org/std/cell/struct.LazyCell.html).


PS you don't need lazy_static/once_cell for this particular case. `Mutex::new` is const from 1.63 and `AtomicBool::new` is const from 1.24.
You can mark `MockBackend::new` as const too.